components/signup/signup_email/signup_email.tsx

```diff
@@ -290,7 +306,7 @@ export default class SignupEmail extends React.PureComponent {
                     defaultMessage='Valid email required for sign-up'
                 />
             </span>
-        );
+        ) as React.ReactNode; /* Assertion used to allow for setting variable to an empty string below */
```

Comment:
Can we avoid the type assertion here if we change the code below to replace `emailHelpText = ''` with `emailHelpText = undefined`? They should be functionally equivalent, and I think the only reason we're using strings here is that it's old code from before we realized passing `null` or `undefined` was better than passing a string for an unrendered element.

Reply:
Typescript says that 'undefined' is not assignable to type JSX.Element. The same thing goes for when I assign emailHelpText to null. I'm going to do some digging and see if I can find a cleaner fix.

Reply:
You can always pass JSX.Element | undefined for conditionally rendered elements.

Reply:
@nevyangelova Are you saying that I should replace `as React.ReactNode` with `as JSX.Element | undefined` and then set emailHelpText to undefined?

Reply:
Perhaps you could define `emailHelpText` as `let emailHelpText: React.ReactNode = (...);` instead then? The differences between the two ways of doing it are subtle, but it gives us more type safety by avoiding `as`. More specifically:
1. `let emailHelpText: React.ReactNode = foo;` tells TS that `emailHelpText` should contain a `React.ReactNode` so it can throw an error if `foo` happens to not be something that matches the type.
2. `let emailHelpText = foo as React.ReactNode;` tells TS that `foo` is definitely a `React.ReactNode`, but that prevents it from throwing a type error if `foo` is the wrong type.

Reply:
Okay, then I'll go with the type annotation instead of the type assertion. Thanks for the explanation, I didn't realize how dangerous type assertions could be.